### PR TITLE
Shrink 'any-generics' whitelist for 'writers' module

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

File: sphinx/writers/latex.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ class LaTeXWriter(writers.Writer):
         ('Document class', ['--docclass'], {'default': 'manual'}),
         ('Author', ['--author'], {'default': ''}),
     ))
-    settings_defaults: Dict = {}
+    settings_defaults: Dict[str, str] = {}
```
Comment (danieleades) [inline sphinx/writers/latex.py]: is this right?

**Comment by danieleades** [comment]
Hi @AA-Turner, i see you changed the base from `5.x` to `master`. Why the change, may I ask?

**Comment by AA-Turner** [comment]
5.x branch is dead, Sphinx is now using a single branch model.

A

File: sphinx/writers/latex.py (hunk 1 above)
Comment (AA-Turner) [inline sphinx/writers/latex.py]:
```suggestion
    settings_defaults: Dict[str, Any] = {}
```

File: sphinx/writers/texinfo.py [hunk 2]
```diff
@@ -113,9 +113,9 @@ class TexinfoWriter(writers.Writer):
             ('Category', ['--texinfo-dir-category'], {'default':
                                                       'Miscellaneous'})))
 
-    settings_defaults: Dict = {}
+    settings_defaults: Dict[str, str] = {}
```
Comment (AA-Turner) [inline sphinx/writers/texinfo.py]:
```suggestion
    settings_defaults: Dict[str, Any] = {}
```

File: sphinx/writers/text.py [hunk 3]
```diff
@@ -357,7 +357,7 @@ def my_wrap(text: str, width: int = MAXWIDTH, **kwargs: Any) -> List[str]:
 class TextWriter(writers.Writer):
     supported = ('text',)
     settings_spec = ('No options here.', '', ())
-    settings_defaults: Dict = {}
+    settings_defaults: Dict[str, str] = {}
```
Comment (AA-Turner) [inline sphinx/writers/text.py]:
```suggestion
    settings_defaults: Dict[str, Any] = {}
```

**Pull request merged**